packages/node_modules/@node-red/editor-client/src/js/ui/common/editableList.js

```diff
@@ -67,24 +68,53 @@
                 this.topContainer.addClass(this.options.class);
             }
 
+            var buttons = this.options.buttons || [];
+
             if (this.options.addButton !== false) {
-                var addLabel;
+                var addLabel, addTittle;
                 if (typeof this.options.addButton === 'string') {
                     addLabel = this.options.addButton
                 } else {
                     if (RED && RED._) {
                         addLabel = RED._("editableList.add");
+                        addTitle = RED._("editableList.addTitle");
                     } else {
                         addLabel = 'add';
+                        addTitle = 'add new item';
                     }
                 }
-                $('<a href="#" class="red-ui-button red-ui-button-small red-ui-editableList-addButton" style="margin-top: 4px;"><i class="fa fa-plus"></i> '+addLabel+'</a>')
-                    .appendTo(this.topContainer)
+                buttons.unshift({
+                    label: addLabel,
+                    icon: "fa fa-plus",
+                    click: function(evt) {
+                        that.addItem({});
+                    },
+                    title: addTitle
+                });
+            }
+
+            buttons.forEach(function(button) {
+                var innerHtml = "";
+                var titleAttribute="";
+                if (button.icon) {
+                    innerHtml = '<i class="'+button.icon+'"></i> ';
```

Comment:
In general we should avoid string concatenation when building up HTML content - it provides a potential route for injecting bad content into the page.

For example, if `button.label` is `<script>alert('hi');</script>`.

Given the values you are joining here aren't going to be provided by the end user, it isn't so critical - but something to keep in mind for the future.

Reply:
Hey Nick,
Indeed it is not critical here, but it is a very good remark in your review!!!  
Have never taken this into account, but at least I want to learn from my mistake...
Like you say, when I use this:
```
buttons: [{
   label: "<script>alert('hi');<\/script>"
}
```
Then the injected code is immediately executed...

Based on this [StackOverflow ](https://stackoverflow.com/a/49396917) discussion it seems that I have to use the Jquery `text()`, instead of inner html.  I have updated my code, and then indeed the injected code isn't being executed anymore.  Instead the injected code is now being displayed as plain text button label:

![image](https://user-images.githubusercontent.com/14224149/108768708-47605700-7558-11eb-9a38-525345fede1e.png)

Reply:
I will need to fix something: by using the `text()` function, the fontAwesome classes are also escaped (as you can see in the first "Add" button of the screenshot above).  To be continued ...

Reply:
I think that now all user input is correctly escaped (via `text()` or `attr()`).

![image](https://user-images.githubusercontent.com/14224149/108781603-7253a680-756a-11eb-8e6e-4fb150ba8b47.png)